ql/cashflows/cashflows.cpp

```diff
@@ -846,24 +851,37 @@ namespace QuantLib {
         if (npvDate == Date())
             npvDate = settlementDate;
 
+        // use original leg if sorted, otherwise create
+        // a sorted copy and use that
+        const Leg *refLeg;
+        CashFlowLess cashFlowLess;
+        Leg tmp;
+        if (std::is_sorted(leg.begin(), leg.end(), cashFlowLess)) {
```

Comment:
This adds another scan of the leg even if it's already sorted. I don't know how much of a performance penalty it is, but I'm not sure that we should add it unconditionally. How about simply documenting the requirement, and possibly adding a `QL_REQUIRE` only when extra checks are turned on?